controller/ca/controller.go

```diff
@@ -0,0 +1,216 @@
+package ca
+
+import (
+	"time"
+
+	"github.com/runconduit/conduit/controller/k8s"
+	pkgK8s "github.com/runconduit/conduit/pkg/k8s"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type CertificateController struct {
+	namespace   string
+	k8sAPI      *k8s.API
+	syncHandler func(key string) error
+	queue       workqueue.RateLimitingInterface
+}
+
+func NewCertificateController(conduitNamespace string, k8sAPI *k8s.API) *CertificateController {
+	c := &CertificateController{
+		namespace: conduitNamespace,
+		k8sAPI:    k8sAPI,
+		queue: workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(), "certificates"),
+	}
+
+	k8sAPI.Pod().Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.handlePodAdd,
+			UpdateFunc: c.handlePodUpdate,
+		},
+	)
+
+	k8sAPI.CM().Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.handleConfigMapAdd,
+			UpdateFunc: c.handleConfigMapUpdate,
+			DeleteFunc: c.handleConfigMapDelete,
+		},
+	)
+
+	c.syncHandler = c.syncNamespace
+
+	return c
+}
+
+func (c *CertificateController) Run(stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	log.Info("starting certificate controller")
+	defer log.Info("shutting down certificate controller")
+
+	go wait.Until(c.worker, time.Second, stopCh)
```

Comment:
curious at the interaction between `wait.Until` and `processNextWorkItem`. 

from https://godoc.org/k8s.io/apimachinery/pkg/util/wait#Until:
`Until loops until stop channel is closed, running f every period.`

the implementation of `worker()` calls `c.queue.Get()` in a for loop. from https://godoc.org/k8s.io/client-go/util/workqueue#Type.Get:
`Get blocks until it can return an item to be processed.`

wouldn't `worker()` run forever on its own, without the need for `wait.Until` wrapping it?